data/trie/patriciaMerkleTrie.go

```diff
@@ -358,6 +358,9 @@ func (tr *patriciaMerkleTrie) ResetOldHashes() [][]byte {
 
 // GetDirtyHashes returns all the dirty hashes from the trie
 func (tr *patriciaMerkleTrie) GetDirtyHashes() (data.ModifiedHashes, error) {
+	tr.mutOperation.Lock()
```

Comment:
Do you need here this mutex lock/unlock?

Reply:
Yes, because every trie operation uses the same mutex, and all trie operations must be serialized.